src/protocols/secure_channel/SessionEstablishmentExchangeDispatch.cpp

```diff
@@ -45,6 +45,17 @@ CHIP_ERROR SessionEstablishmentExchangeDispatch::SendMessageImpl(SecureSessionHa
     return CHIP_ERROR_INCORRECT_STATE;
 }
 
+CHIP_ERROR SessionEstablishmentExchangeDispatch::ResendMessage(SecureSessionHandle session, EncryptedPacketBufferHandle && message,
+                                                               EncryptedPacketBufferHandle * retainedMessage) const
+{
+    if (mTransportMgr != nullptr)
+    {
+        return mTransportMgr->SendMessage(mPeerAddress, std::move(message));
```

Comment:
Don't we need to set `*retainedMessage` somewhere here?  Otherwise only one resend will work and the second will fail.  Please add a test for multiple resends.

Reply:
Updated the code, and added a test to TestPASESession, where it drops two consecutive messages.